diff -rupN a/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp b/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp
--- a/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp	2009-03-02 09:08:49.000000000 +1100
+++ b/cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp	2009-11-05 19:40:16.000000000 +1100
@@ -131,10 +131,10 @@ namespace CEGUI
         {
             switch(childNode->Type())
             {
-            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::ELEMENT:
+            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::TINYXML_ELEMENT:
                 processElement(childNode->ToElement());
                 break;
-            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::TEXT:
+            case CEGUI_TINYXML_NAMESPACE::TiXmlNode::TINYXML_TEXT:
                 if (childNode->ToText()->Value() != '\0')
                     d_handler->text((utf8*)childNode->ToText()->Value());
                 break;
