From 74ed81c79c37c852d9011159efe1ddbdecb1fb65 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 02 Jun 2010 22:30:12 +0000
Subject: [terminal] Add some debug spew if terminal couldn't be opened

---
diff --git a/src/libply-splash-core/ply-terminal.c b/src/libply-splash-core/ply-terminal.c
index ff59d71..6140fa4 100644
--- a/src/libply-splash-core/ply-terminal.c
+++ b/src/libply-splash-core/ply-terminal.c
@@ -462,7 +462,10 @@ ply_terminal_open_device (ply_terminal_t *terminal)
   terminal->fd = open (terminal->name, O_RDWR | O_NOCTTY);
 
   if (terminal->fd < 0)
-    return false;
+    {
+      ply_trace ("Unable to open terminal device '%s': %m");
+      return false;
+    }
 
   terminal->fd_watch = ply_event_loop_watch_fd (terminal->loop, terminal->fd,
                                                    PLY_EVENT_LOOP_FD_STATUS_NONE,
--
cgit v0.8.3-6-g21f6
